### PR TITLE
Add 3.9 rules

### PR DESCRIPTION
**Pull request by gousaiyang**
- Also include some corrections
- Also include some pre-3.9 rules (mainly because there were documented later than 3.8)

Note: the complete `BUILTIN_GENERIC_ANNOTATION_TYPES` list is taken from [`Lib/test/test_genericalias.py` in CPython repo](https://github.com/python/cpython/blob/v3.9.0/Lib/test/test_genericalias.py#L48). This list may grow in the future, so we may need to "watch" this test file and update the rules in the future.

### CODE REVIEW

File: tests/function.py [hunk 1]
```diff
@@ -3830,12 +3832,6 @@ def test_upgrade_dependencies_from_venv_EnvBuilder(self):
   def test_issoftkeyword_from_keyword(self):
     self.assertOnlyIn((3, 9), detect("from keyword import issoftkeyword"))
 
-  def test_reversed_from_types_MappingProxyType(self):
-    self.assertOnlyIn((3, 9),
-                      detect("from types import MappingProxyType\n"
-                             "mpt = MappingProxyType({})\n"
-                             "mpt.reversed()"))
```
Comment (netromdk) [inline tests/function.py:3837]: Why was this removed?

https://docs.python.org/3.9/library/types.html?highlight=reversed#types.MappingProxyType

File: tests/module.py [hunk 2]
```diff
@@ -427,8 +430,8 @@ def test_zipapp(self):
   def test_urlparse(self):
     self.assertOnlyIn((2, 0), detect("import urlparse"))
 
-  def test_timezone(self):
-    self.assertOnlyIn((3, 9), detect("import timezone"))
+  def test_zoneinfo(self):
+    self.assertOnlyIn((3, 9), detect("import zoneinfo"))
```
Comment (netromdk) [inline tests/module.py:434]: Hah.. Thanks for spotting this typo!

File: vermin/rules.py [hunk 3]
```diff
@@ -3199,10 +3267,13 @@ def MOD_MEM_REQS():
 }
 
 # Builtin types that work as annotation types from 3.9+
+# The list is taken from Lib/test/test_genericalias.py in the CPython repository
```
Comment (netromdk) [inline vermin/rules.py:3270]: It would be nice to add the URL so that it's very easy to find: https://github.com/python/cpython/blob/master/Lib/test/test_genericalias.py

File: tests/function.py [hunk 4]
```diff
@@ -3850,3 +3846,12 @@ def test_send_fds_from_socket(self):
 
   def test_recv_fds_from_socket(self):
     self.assertOnlyIn((3, 9), detect("from socket import recv_fds"))
+
+  def test_create_module_from_importlib_machinery_FrozenImporter(self):
+    self.assertOnlyIn((3, 4), detect("import importlib.machinery.FrozenImporter\nimportlib.machinery.FrozenImporter.create_module()"))
+
+  def test_exec_module_from_importlib_machinery_FrozenImporter(self):
+    self.assertOnlyIn((3, 4), detect("import importlib.machinery.FrozenImporter\nimportlib.machinery.FrozenImporter.exec_module()"))
```
Comment (netromdk) [inline tests/function.py:3854]: This line is too long. Please run `make check-all` and you'll probably find other cases.

For some reason, my TravisCI checks didn't execute for your PR..

**Review by netromdk — CHANGES_REQUESTED** [review]
Thanks a lot for these rules and fixes! I've only a few minor comments.

File: tests/function.py (hunk 1 above)
Comment (gousaiyang) [reply tests/function.py:3837]: Because the fact is that the builtin function `reversed()` now accepts `MappingProxyType` objects as arguments, rather than the `MappingProxyType` having a new `reversed` method.

Comment (netromdk) [reply tests/function.py:3837]: Ah, right. That makes sense.

**Comment by netromdk** [comment]
Weird, the checks were run but not shown here like they normally are: https://travis-ci.org/github/netromdk/vermin/builds/734362918

**Review by netromdk — APPROVED** [review]
Perfect!
